### PR TITLE
Sound refactoring

### PR DESCRIPTION
**Pull request by Chleba**
A little refactoring was needed, when getting feedback from testing department for better adding new sounds and defining it's values with less code needed.

BFW-977

### CODE REVIEW

File: src/common/sound.cpp [hunk 1]
```diff
@@ -14,6 +14,34 @@ Sound::Sound() {
     frequency = 100.f; // frequency of sound signal (0-1000)
     volume = 0.50;     // volume of sound signal (0-1)
 
+    // -- values of sound signals - frequencies, volumes, durations
+    // -- durations of beep
+    durations[eSOUND_TYPE_ButtonEcho] = 100.f;
```
Comment (DRracer) [inline src/common/sound.cpp:19]: I'd recommend doing some static initialization. The compiler might understand your code and optimize it away completely, but I wouldn't rely on it.
`static const float durations[] = {100.F, 500.F ....};`
or even better - make a struct and initialize all components together (will be more readable)
`struct DFV { float duration, freq, volume; }; DFV dfv[] = { { 100.F, 900.F, volume }, { .... } };` if volume is a `static const float volume = 0.5F`

File: src/common/sound.cpp [hunk 2]
```diff
@@ -14,6 +14,34 @@ Sound::Sound() {
...
+    durations[eSOUND_TYPE_StandardPrompt] = 500.f;
```
Comment (DRracer) [inline src/common/sound.cpp:20]: oh, durations are uin32_t, why do you assign floating point numbers?

File: src/common/sound.cpp [hunk 3]
```diff
@@ -14,6 +14,34 @@ Sound::Sound() {
...
+    durations[eSOUND_TYPE_StandardAlert] = 200.f;
+    durations[eSOUND_TYPE_CriticalAlert] = 500.f;
+    durations[eSOUND_TYPE_EncoderMove] = 50.f;
+    durations[eSOUND_TYPE_BlindAlert] = 100.f;
+    durations[eSOUND_TYPE_Start] = 100.f;
+
+    // -- frequencies of beep
+    frequencies[eSOUND_TYPE_ButtonEcho] = 900.f;
+    frequencies[eSOUND_TYPE_StandardPrompt] = 600.f;
```
Comment (DRracer) [inline src/common/sound.cpp:29]: frequencies are double? Why? Double is not HW accelerated on our CPU? Assigning a float into a double?

File: src/common/sound.h [hunk 4]
```diff
@@ -30,6 +30,21 @@ class Sound {
     double frequency;
     double volume;
 
+    uint32_t durations[eSOUND_TYPE_count];
```
Comment (DRracer) [inline src/common/sound.h:33]: use reasonable data types and initialize them properly

File: src/common/sound.cpp [hunk 5]
```diff
@@ -51,125 +79,42 @@ void Sound::stop() {
...
-            this->soundEncoderMove(1, 50.f);
-        }
-        if (eSoundType == eSOUND_TYPE_BlindAlert) {
-            this->soundBlindAlert(1, 100.f);
-        }
-        if (eSoundType == eSOUND_TYPE_CriticalAlert) {
-            this->soundCriticalAlert(-1, 500.f);
-        }
+        t_size = sizeof(assistTypes) / sizeof(assistTypes[0]);
+        this->_playSound(eSoundType, assistTypes, assistRepeats, t_size);
```
Comment (DRracer) [inline src/common/sound.cpp:108]: why do you use explicit `this->` ?

File: src/common/sound.cpp (hunk 1 above)
Comment (DRracer) [reply src/common/sound.cpp:19]: Also, when you write comments (which is good), please use the doxygen style and prefer documenting the public interface methods in the header file (their purpose and meaning of parameters).
Implementation-related comments are ok to put near the implementation (i.e. into the cpp file)

File: src/common/sound.cpp (hunk 2 above)
Comment (Chleba) [reply src/common/sound.cpp:20]: my mistake - i copied previous frequency values and compiler did not give a damn about that.

File: src/common/sound.cpp (hunk 3 above)
Comment (Chleba) [reply src/common/sound.cpp:29]: Another mistake of not checking my types. Fixed.

File: src/common/sound.cpp (hunk 5 above)
Comment (Chleba) [reply src/common/sound.cpp:108]: bad habit from previous languages - fixed

File: src/common/sound.cpp (hunk 1 above)
Comment (Chleba) [reply src/common/sound.cpp]: Doxygen commentary studied and implemented and static defining of signal's values changed.

File: src/common/sound.h (hunk 4 above)
Comment (Chleba) [reply src/common/sound.h]: done

File: src/guiapi/include/MItem_tools.hpp [hunk 6]
```diff
@@ -216,14 +216,17 @@ class MI_SOUND_MODE : public WI_SWITCH_t<4> {
...
-class MI_SOUND_TYPE : public WI_SWITCH_t<5> {
-    constexpr static const char *const label = N_("Sound Type");
-
-    constexpr static const char *str_ButtonEcho = N_("ButtonEcho");
-    constexpr static const char *str_StandardPrompt = N_("StandardPrompt");
-    constexpr static const char *str_StandardAlert = N_("StandardAlert");
-    constexpr static const char *str_EncoderMove = N_("EncoderMove");
-    constexpr static const char *str_BlindAlert = N_("BlindAlert");
+class MI_SOUND_TYPE : public WI_SWITCH_t<8> {
+    constexpr static const char *const label = "Sound Type";
```
Comment (DRracer) [inline src/guiapi/include/MItem_tools.hpp:220]: translations!

**Pull request merged**